bindings/nodejs/lib/types/block/id.ts

```diff
@@ -2,6 +2,41 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { HexEncodedString } from '../utils';
+import { SlotIndex } from './slot';
+
+/**
+ * Base class for IDs with a hex encoded slot index at the end.
+ */
+export class IdWithSlotIndex {
+    id: string;
+    constructor(id: string) {
+        this.id = id;
+    }
+    inspect() {
+        return this.id;
+    }
+    toString() {
+        return this.id;
+    }
+    toJSON() {
+        return this.id;
+    }
```

Comment:
Is there some way to have a class without named field? Haven't found a way

Reply:
Did you mean to do something like this?
```typescript
class IdWithSlotIndex extends String {
    slotIndex(): SlotIndex {
        const numberString = super.toString().slice(66, 74);
        const chunks = [];
        for (
            let i = 0, charsLength = numberString.length;
            i < charsLength;
            i += 2
        ) {
            chunks.push(numberString.substring(i, i + 2));
        }
        const separated = chunks.map((n) => parseInt(n, 16));
        const buf = Uint8Array.from(separated).buffer;
        const view = new DataView(buf);
        return view.getUint32(0, true);
    }
}
```

Reply:
Maybe extend the String class and just add a method (tried this in devtools, seams to work)
```
class IdWithSlotIndex extends String {
    slotId() {
        this.length > 8 ? this.slice(-8) : this
    }
}
```
This way when you instantiate a "string" with `new IdWithSlotIndex("someBlockId")` you get the capabilities of the added function

Reply:
Yes, thanks!

Reply:
@msarcev but don't you see this as problem if it's available for all strings? Because most strings don't have a slot index encoded at the end

Reply:
Ah, maybe you mean that you can put any string in there, even ones that are not "blockId/transactionId" 🤔 
That was also the case with your first solution though.
@VmMad suggested that we could add checks (maybe in the constructor) that assert if you passed something like a blockId (maybe by length). This would also mean handling errors if construction of that thing fails.
Wdyt ?

Reply:
We could do something like:
```
export class IdWithSlotIndex extends String {
    private _isValid: boolean = true;

    constructor(id: string) {
        super(id)

        if (id.length < 64) {
            this._isValid = false;
        }
    }

    slotIndex(): SlotIndex {
        // ...
    }

    get isValid() {
        return this._isValid
    }
}
```
Then you can use it like:
const blockId = new IdWithSlotIndex("123123123")
```
if (blockId.isValid) {
    // use it
}
```

Reply:
@Thoralf-M ☝🏻

Reply:
> It won’t be “available for all strings”, only strings that are instantiated with new IdWithSlotIndex(… , no ?

Yes, sorry misread something

Sounds weird to me to have an `isValid()` method
But do I get it right that you prefer to use `new IdWithSlotIndex("0x...").slotIndex()` over `new BlockId("0x...").slotIndex()` if `BlockId` stays a type for a string instead of a class?

Reply:
Maybe combine everything and instead of even having this `IdWithSlotIndex` class, make all "string" types that have a slot index encoded be a subclass of `String` with the added method explicitly.
```
class BlockId extends String {
    slotIndex() {
        // call to shared util
    }
}

class TransactionId extends String {
    slotIndex() {
        // call to shared util
    }
}
```
This way we can construct with `new BlockId("123123")` fine